### PR TITLE
Better networking code

### LINKED ISSUE #344: Network bandwidth overload
When having just Flaxbeard'sSteamPower-1.7.10-0.28.12 and Forge, just connecting to a Vanilla server, my client upload bandwidth raise to 10 kB/s.
When I do the same without the mod, I'm seating around 1 kB/s.
Connecting to an server with the mod doesn't change the result.

According to Opis, the related packet is:
C17PacketCustomPayload 0 41,00 packet/s 10,573 KiB/s 936,769 KiB

I've seen a lot of changes in the code since Nov 7th version. Could we get a more recent binary?

### PR DESCRIPTION
**Pull request by elifoster**
This PR does quite a lot.

Firstly, it completely rewrites all of the networking code in the mod. The mod used an ancient packet system using gigantic switch-case statements and just generally terrible methods. With this PR, this code has been replaced by SimpleImpl/SimpleNetworkWrapper networking code.

Secondly, it removes a lot of the packets that were made seemingly on a whim without much thought. These packets were typically just to modify or read NBT data. Doing this each tick, which is what we were doing, causes a massive bandwidth issue. These packets are:
* NoSpacePacket, which was called in every tick that the player did not hold the jump keybind. This was essentially every single tick that the player wasn't jumping. Yeah. 
* SpacePacket, which was called in every tick that the player held the jump keybind.
* RocketJumpHackyPacket, which was called when the rockets exploded.
* ExplodePacket, which was called never
* GrapplePacket, which was called when the player used the Piton Deployer successfully.

Thirdly, it fixes some bugs that could occur caused by modifying NBT on the client-side rather than the server-side.

Fourth, it improves some styling, and fixes some minor issues, along the way. The miscellaneous bugs fixed are:
* One of the Hammer Slots never let items in it, which could have caused some issues.
* Replace isPressed() calls with getIsKeyPressed() from #342.

This PR closes #344 because we now only have 2 packets, neither which are created every single tick like the old ones.

### CODE REVIEW

**Comment by elifoster** [comment]
@xbony2 or @Strikingwolf can one or both of you review this? I'd like to get it merged within a day or two.

File: src/main/java/flaxbeard/steamcraft/handler/SteamcraftTickHandler.java [hunk 1]
```diff
@@ -33,6 +35,86 @@
...
     private int zoomSettingOn = 0;
     private boolean lastPressingKey = false;
+    private boolean isJumping = false;
+
+    @SubscribeEvent
+    public void onPlayerTick(TickEvent.PlayerTickEvent event) {
+        EntityPlayer player = event.player;
+        World world = player.worldObj;
+        boolean isClient = event.side == Side.CLIENT;
+        boolean isServer = event.side == Side.SERVER;
```
Comment (xbony2) [inline src/main/java/flaxbeard/steamcraft/handler/SteamcraftTickHandler.java]: Don't you really just need one boolean here? :c I mean, it's either client or server, it's never both as far as I know.

Comment (elifoster) [inline src/main/java/flaxbeard/steamcraft/handler/SteamcraftTickHandler.java]: Yes. I was going for readability/explicitness. I guess `!isClient` or `!isServer` would be fine though.

**Comment by xbony2** [comment]
No other comments here.

**Pull request merged**